lib/models/ColdStorageStatusQueueEntry.js

```diff
@@ -4,15 +4,15 @@ const entrySchema = joi.object({
     op: joi.string().valid('restore', 'delete', 'archive').required(),
     bucketName: joi.string().required(),
     objectKey: joi.string().required(),
-    objectVersion: joi.string().required(),
+    objectVersion: joi.string(),
     accountId: joi.string().required(),
     archiveInfo: joi.object().required(),
     requestId: joi.string().required(),
 });
 
 class ColdStorageStatusQueueEntry {
     static createFromKafkaEntry(kafkaEntry) {
-        const res = entrySchema.validate(kafkaEntry.value);
+        const res = entrySchema.validate(JSON.parse(kafkaEntry.value));
 
         if (res.error) {
             return {
```

Comment:
The error returned is not really "malformed json" as that would make `JSON.parse` throw an exception (which needs to be caught btw). It's more "validation failed".

Reply:
I see. I had added `JSON.parse` when testing against string inputs. It should have been removed when the testing with `Buffer` type.